### DISCUSSION TITLE
ContextMenu右击菜单选中item与实际selectItem不同

### DISCUSSION THREAD

**Discussion by RickyHe7** [BYJRK/DotNet-Discussions#208] [WPF 开发]
遇到一个问题，在用第三方一个ItemControl控件syncfusion:TreeNavigator时，右键的时候并不是默认选定当前项的，必须先选中指定行，再右键才可以。各位大佬有什么解决办法吗

**Comment by BYJRK** [comment]
你用的是什么第三方控件？贴一个链接吧

**Reply by RickyHe7** [reply]
https://help.syncfusion.com/wpf/tree-navigator/getting-started

**Reply by RickyHe7** [reply]
Nuget 里就有

**Comment by xingzuhui** [comment]
这个好理解,就是他右键弹出的菜单的命令,有一个参数,参数传回的不是当前行,而是isselected的行,其实这样做没错,但是最好修改方法是,在item上右键的时候,默认选中当前行,如果没按ctlrl 和shift, 就和左键一样,触发器也可以,就可以了.不是修改传回选中行参数.
另外说下,这个选中行的参数可以是list,但是他不是泛型集合.也是不支持linq的好像.我操作项只能foreach .

**Reply by xingzuhui** [reply]
我说的是多选的时候

**Comment by SlimeNull** [comment]
我猜，你是想做一个列表项的指令，但是该指令依赖于该项数据，于是你想取到已选择的Item，但发现右键时，并不会选中它，对吗？

如果是这样的话，那这又是一个经典的集合绑定问题。搜索一下 “WPF BindingProxy”，StackOverflow 会有人告诉你这该如何解决。
